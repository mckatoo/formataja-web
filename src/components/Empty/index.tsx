import Button from 'components/Button'
import Link from 'next/link'
import * as S from './styles'

export type EmptyProps = {
  title: string
  description: string
  link?: {
    src: string
    label: string
  }
}

const Empty = ({ title, description, link }: EmptyProps) => (
  <S.Wrapper>
    <S.Image
      src="/img/empty.svg"
      alt="A user who did not find what he was looking for."
      role="image"
    />

    <S.Title>{title}</S.Title>

    <S.Description>{description}</S.Description>

    {!!link && (
      <Link href={link.src} passHref>
        <Button as="a">{link.label}</Button>
      </Link>
    )}
  </S.Wrapper>
)

export default Empty
