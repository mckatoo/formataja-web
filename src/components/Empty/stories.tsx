import { Story, Meta } from '@storybook/react/types-6-0'

import Empty, { EmptyProps } from '.'

export default {
  title: 'Empty',
  component: Empty,
  parameters: {
    backgrounds: {
      default: 'ikatoo-dark'
    }
  }
} as Meta

export const Basic: Story<EmptyProps> = (args) => <Empty {...args} />
Basic.args = {
  title: 'alsdjflasdkfjsdfj',
  description: 'sldfj slkdfj sdklfj sdl'
}

export const WithLink: Story<EmptyProps> = (args) => <Empty {...args} />
WithLink.args = {
  title: 'alsdjflasdkfjsdfj',
  description: 'sldfj slkdfj sdklfj sdl',
  link: {
    src: '/',
    label: 'Go back to store.'
  }
}
