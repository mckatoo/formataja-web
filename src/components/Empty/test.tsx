import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Empty from '.'

export const props = {
  title: 'LKJfklsdjfklsdfkjsdfklsdj askldfjskldfj',
  description: 'Aaklsdjf alskdjflksdf sdlkf'
}

describe('<Empty />', () => {
  it('should render corectly without link', () => {
    renderWithTheme(<Empty {...props} />)

    expect(
      screen.getByRole('image', {
        name: /A user who did not find what he was looking for./i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(screen.getByText(props.description)).toBeInTheDocument()

    expect(screen.queryByRole('button')).not.toBeInTheDocument()
  })

  it('should render with link', () => {
    const link = {
      link: {
        src: '/',
        label: 'Go back to docs.'
      }
    }
    const { container } = renderWithTheme(<Empty {...props} {...link} />)

    expect(
      screen.getByRole('link', { name: /go back to docs/i })
    ).toHaveAttribute('href', '/')
    expect(container.parentElement).toMatchSnapshot()
  })
})
