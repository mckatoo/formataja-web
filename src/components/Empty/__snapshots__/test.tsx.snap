// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render with link 1`] = `
.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#f01320 0%,#ff626b 50%);
  color: #EADEDA;
  cursor: pointer;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c4:hover {
  background: linear-gradient(180deg,#D2121E 0%,#E27B81 50%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  max-width: 100%;
}

.c2 {
  color: #EF2D56;
  font-size: 2.8rem;
}

.c3 {
  color: #EADEDA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

<body>
  <div>
    <main
      class="c0"
    >
      <img
        alt="A user who did not find what he was looking for."
        class="c1"
        role="image"
        src="/img/empty.svg"
      />
      <h2
        class="c2"
      >
        LKJfklsdjfklsdfkjsdfklsdj askldfjskldfj
      </h2>
      <p
        class="c3"
      >
        Aaklsdjf alskdjflksdf sdlkf
      </p>
      <a
        class="c4"
        href="/"
      >
        <span>
          Go back to docs.
        </span>
      </a>
    </main>
  </div>
</body>
`
